packages/runtime/src/helpers/functions.ts

```diff
@@ -208,6 +225,143 @@ export const setupImageFunction = async ({
   }
 }
 
+const traceRequiredServerFiles = async (publish: string): Promise<string[]> => {
+  const requiredServerFilesPath = join(publish, 'required-server-files.json')
+  const {
+    files,
+    relativeAppDir,
+    config: {
+      experimental: { outputFileTracingRoot },
+    },
+  } = (await readJSON(requiredServerFilesPath)) as {
+    files: string[]
+    relativeAppDir: string
+    config: { experimental: { outputFileTracingRoot: string } }
+  }
+  const appDirRoot = join(outputFileTracingRoot, relativeAppDir)
+  const absoluteFiles = files.map((file) => join(appDirRoot, file))
+
+  absoluteFiles.push(requiredServerFilesPath)
+
+  return absoluteFiles
+}
+
+const traceNextServer = async (publish: string): Promise<string[]> => {
+  const nextServerDeps = await getDependenciesOfFile(join(publish, 'next-server.js'))
+
+  // during testing, i've seen `next-server` contain only one line.
```

Comment:
This feels like a weird check?

What if Next.js itself changes the file and the .nft file changes correctly and this arbitrary threshold is triggered? What is the goal of this check?

Reply:
I've explained that here: https://github.com/netlify/next-runtime/pull/2058#discussion_r1187332827

Rephrasing, to add more context: I've seen instances locally where the `.nft` file contained less than 10 lines, but those were definitely invalid ones, and they resulted in broken bundles. I don't see a case where the `.nft` file *correctly* contains less than 10 lines. I'm unsure if the instances i've seen locally are due to not-so-common setup where the plugin is pulled in with `npm link`. If this happened in prod, it'd result in deployments that break at runtime. To prevent this, i'm adding this build-time check. I'll be closely monitoring it during the rollout. Once I'm certain this doesn't occur in production, I'll remove the check.